CMake FindX alterations

Phatcat opened the pull request:
Marking all as advanced without doing an if on PACKAGE_FOUND and nullifying the include_dir and library if not found will hide the entry in cmake gui on windows. Adding an entry and pointing to the correct path at this point will not work so you are forced to do a hard reset of the cache, so you have to enter every single entry exactly right before selecting 'configure' or it will fail and you'll have to do a hard reload of the cache and start from scratch, which can become very annoying.

With these changes you can open the project in CMake and press configure right away and then simply start replacing the NOTFOUND paths with the actual paths and simply press 'configure' again, no need for any hard reload of cache.

Also there's no need to hide BOTAN_ROOT_DIR and MYSQLCCPP_ROOT_DIR at all, just like PCRE_INCLUDE_DIR and PCRE_LIBRARY shouldn't be hidden either.

Phatcat committed:
CMake FindX alterations

arking all as advanced without doing an if on PACKAGE_FOUND and nullifying the include_dir and library if not found will hide the entry in cmake gui on windows. Adding an entry and pointing to the correct path at this point will not work so you are forced to do a hard reset of the cache, so you have to enter every single entry exactly right before selecting 'configure' or it will fail and you'll have to do a hard reload of the cache and start from scratch, which can become very annoying.

With these changes you can open the project in CMake and press configure right away and then simply start replacing the NOTFOUND paths with the actual paths and simply press 'configure' again, no need for any hard reload of cache.

Also there's no need to hide BOTAN_ROOT_DIR and MYSQLCCPP_ROOT_DIR at all, just like PCRE_INCLUDE_DIR and PCRE_LIBRARY shouldn't be hidden either.

Phatcat committed:
Create FindZlib.cmake

Don't use build in cmake Findzlib since it doesn't actually fail on zlib not found and then it hides the ZLIB_INCLUDE_DIR and ZLIB_LIBRARY so you have to do a hard reload of cmake to get it working. Adding a FindZlib and doing the version control here as well as make sure to display the entry in cmake gui

Phatcat committed:
More CMake FindX alterations

Both Botan and MySQLConnector-C++ when build in debug does not create specific debug libraries, so they cannot exist in the same folder at the same time, unlike pcre and zlib which create their debug libraries as specific debug.

Therefor I've made even more drastic alterations to the FindX cmake files.

Phatcat committed:
Update FindPCRE.cmake

FindPCRE.cmake was not looking for and comparing the version number making the required version set in CMakeList.txt unutilized.

Chaosvex commented:
Thanks. I'll have to review this quite carefully. 

One thing that jumps out at me is the zlib change. Why the addition of a new file when CMake ships with its own zlib package support, which, so far has worked fine?

Phatcat commented:
To be quite honest it's mostly about preferences I guess. The FindZLIB.cmake that is with cmake modules is a somewhat messy solution in my opinion, utilizing search functionality and looking in hardcoded paths (where it's extremely unlikely to ever be found)

Also I wanted the control of what gets marked as advanced, and the naming scheme, since everywhere else we can call it PROJECT_ROOT_DIR, like BOTAN_ROOT_DIR and MYSQLCCPP_ROOT_DIR, etc. while cmake's built-in findzlib is simply ZLIB_ROOT (which is even inconsistent internally since they name it ZLIB_INCLUDE_DIR), I have tried to make it all a bit more 'uniform' in their behavior and naming, for consistency's sake, advance anything but the root folder is what I set them all to, since the root folder is all that's really needed for each library, which the changes I have proposed (and following a bit of staging set forth in the windows install guide I have pr'd). - Basically I guess there's not any functionality change to the FindZlib I've proposed in comparison in particular besides some semantic/aesthetic/design changes.

Furthermore the version control in cmake's own findZlib looks to not actually be written for zlib, they are called 'major', 'minor' and 'patch', they are 'major', 'minor', 'revision' and 'subrevision'. Not that I think the version control in cmake's findzlib doesn't work, it just doesn't seem to look for subrevision, so even that seems incomplete/sloppy to me.

I also left it as it's own commit intentionally in the case you wanted to separate it from the other changes so it's easy enough to just drop. If it is however dropped I'd like the chance to adjust the remaining FindX.cmake to match in naming scheme.

Phatcat committed:
Update FindBotan.cmake

Correctly identify botan's include dir structure instead of having the user move around files. I guess the structure has been like this since... Botan entered majr version 2 I'm guessing.
